PLUGINS.md

```diff
@@ -0,0 +1,180 @@
+# OpenShift Console Plugins
+
+Openshift Console Plugins extend the functionality of the console. Plugins are enabled by
+adding them as `dependencies` of the [console-app](frontend/packages/console-app) package.
+
+See the [demo plugin](frontend/packages/console-demo-plugin) for reference.
+
+## Directory and File Names
+
+Plugins packages should be in [frontend/packages](frontend/packages) directory. A plugin should have
+a `package.json`, `OWNERS`, and a plugin entry file, the plugin entry file is usually
```

Comment:
Please call them "plugin entry modules" since they are effectively ECMAscript modules.

Having `package.json` and `OWNERS` files is mandatory.

Every plugin should follow the recommended file structure:
- `src` directory
- `integration-tests` directory (if needed)
- plugin entry module at `src/plugin.ts`
- `src/components` for React components, `src/models` for k8s model definitions
- unit tests co-located at `/path/to/unit/__tests__/unit.spec.ts`

In addition, every plugin should typically provide (at least) these two extensions:
- `ModelDefinition` - add new k8s model definitions
- `FeatureFlag/Model` (in future also `FeatureFlag/Action`) - add new feature flag

and use the newly added feature flags (possibly together with core Console flags) to gate its extensions.